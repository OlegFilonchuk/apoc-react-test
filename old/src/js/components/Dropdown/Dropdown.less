@import '~chameleon/less/variables.less';

@disabled-text-color: #B2B2B2;

.dropdown-button-container {
  display: inline-block;
}

.dropdown-option-disabled {
  pointer-events: none;
  opacity: 0.5;
}

.sc-dropdown.sc-dropdown--fullwidth {
  margin-right: 10px;

  .sc-dropdown-menus {
    width: 100%;
    left: 0;
  }

  .sc-btn.sc-btn-dropdown {
    font-family: @font-semibold;

    width: 100%;
    position: relative;
    text-align: left;
    margin: 0;

    .sc-icon-down-arrow {
      left: auto;
      right: 10px;
      position: absolute;
      top: 10px;
    }
  }
}

// DropdownOptionLink icon related styles
.dropdown-option,
.dropdown-option-link {
  .with-icon {
    padding-left: 5px;

    .icon {
      padding: 0 5px;
    }
  }
}

li.dropdown-option-link.disabled a {
  color: @disabled-text-color;
  cursor: not-allowed;
}

.dropdown-filter {
  position: relative;
  padding: (@padding-sm-vertical) (@padding-sm-horizontal * 2);

  &__input {
    width: 100%;
    padding-right: (@padding-base-horizontal * 3);
  }

  .dropdown-filter__icon {
    position: absolute;
    top: @padding-sm-vertical;
    right: (@padding-sm-horizontal * 3);
    font-size: @font-size-xl;
    color: @primary;
  }
}
