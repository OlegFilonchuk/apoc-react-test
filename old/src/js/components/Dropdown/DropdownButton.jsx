import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

export default function DropdownButton({
  children,
  className,
  onClick,
  isExpanded,
  id,
  onKeyDown,
  buttonType,
  ...restProps
}) {
  return (
    <Button
      className={`sc-btn sc-btn-dropdown sc-btn-primary-outline ${className}`}
      type={buttonType}
      aria-haspopup="true"
      aria-expanded={isExpanded}
      onClick={e => onClick(e)}
      id={id}
      onKeyDown={onKeyDown}
      {...getDataTestElementProps(restProps, dataTestElementValue => dataTestElementValue || id)}
      {...restProps}
    >
      <span className="sc-btn-dropdown-text">{children}</span>
      <span className="sc-icon-down-arrow" />
    </Button>
  );
}

DropdownButton.propTypes = {
  /**
   * Text or element inside of a button
   */
  children: PropTypes.node,

  /**
   * Optional callback. Needed by [Dropdown](#dropdown) but custom callback can be set here without any conflict.
   */
  onClick: PropTypes.func,

  /**
   * Optional callback. Needed by [Dropdown](#dropdown) but custom callback can be set here without any conflict.
   */
  onKeyDown: PropTypes.func,

  /**
   * Controls aria-expanded attribute of a button. Optional. Needed by [Dropdown](#dropdown).
   */
  isExpanded: PropTypes.bool,

  /**
   * Sets `id` attribute of a `button`. Useful to match it with a `label`'s `for` attribute to enhance UX.
   */
  id: PropTypes.string,

  /**
   * Css classes of button.
   */
  className: PropTypes.string,

  /**
   * Button type
   */
  buttonType: PropTypes.string,

  ...dataTestElementPropTypes
};

DropdownButton.defaultProps = {
  className: 'sc-btn-default',
  id: '',
  children: undefined,
  isExpanded: false,
  buttonType: 'button',
  onClick: () => true,
  onKeyDown: () => true
};
