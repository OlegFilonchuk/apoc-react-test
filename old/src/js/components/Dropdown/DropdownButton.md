Purpose of this component is to be used together with [Dropdown](#dropdown).

## Standalone example

    <DropdownButton label="Button example"/>
