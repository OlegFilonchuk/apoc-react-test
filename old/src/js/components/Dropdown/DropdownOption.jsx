import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

export default function DropdownOption({
  label,
  name,
  selected,
  value,
  onClick,
  disabled,
  className,
  showPopupOnHover,
  icon: iconClassName,
  ...restProps
}) {
  const classes = classNames({
    'sc-focus': selected,
    'dropdown-option-disabled': disabled,
    'with-icon': iconClassName
  });

  const icon = <span className={`${iconClassName} icon`} />;
  const title = showPopupOnHover ? label : undefined;

  return (
    <li className={`dropdown-option ${className}`}>
      <label className={classes} title={title}>
        {icon}
        {label}
        <input
          type="radio"
          role="menu"
          name={name}
          value={value}
          defaultChecked={selected}
          readOnly="readOnly"
          onClick={onClick}
          disabled={disabled}
          {...getDataTestElementProps(restProps, testElementValue => testElementValue || `option-${value}`)}
        />
      </label>
    </li>
  );
}

/**
 * @typedef {{label: string, name: string, value: string, isChecked: boolean, onClick: Function}} DropdownOptionProps
 */
DropdownOption.propTypes = {
  /**
   * Text or element to be displayed
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,

  /**
   * Sets `input`s `name` attribute. It's not required since [Dropdown](#dropdown)'s prop `name` should do it.
   */
  name: PropTypes.string,

  /**
   * Sets `value` attribute to `input`.
   */
  value: PropTypes.string.isRequired,

  /**
   * Controls css class of a `label` element
   */
  selected: PropTypes.bool,

  /**
   * Ignored options are grayed-out and disabled
   */
  disabled: PropTypes.bool,

  /**
   * Sets onClick on `input` element
   */
  onClick: PropTypes.func,

  icon: PropTypes.string,

  /**
   * Any custom class names
   */
  className: PropTypes.string,

  /**
   * Flag to defined should popup be shown on hover or not
   */
  showPopupOnHover: PropTypes.bool,

  ...dataTestElementPropTypes
};

DropdownOption.defaultProps = {
  name: '',
  disabled: false,
  selected: false,
  onClick: () => true,
  icon: '',
  className: '',
  showPopupOnHover: false
};
