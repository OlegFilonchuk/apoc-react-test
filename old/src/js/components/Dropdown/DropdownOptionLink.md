Purpose of this component is to be used together with [Dropdown](#dropdown). See example there.
