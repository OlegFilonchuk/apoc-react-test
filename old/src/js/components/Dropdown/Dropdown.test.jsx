import React from 'react';
import PropTypes from 'prop-types';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dropdown from './Dropdown';
import DropdownOption from './DropdownOption';
import DropdownOptionLink from './DropdownOptionLink';
import DropdownButton from './DropdownButton';

describe('<Dropdown>', () => {
  describe('custom props types validators', () => {
    describe('children', () => {
      describe('of wrong type', () => {
        it('returns error', () => {
          const validationResult = Dropdown.propTypes.children({ children: <div>test</div> });

          expect(validationResult).toEqual(jasmine.any(Error));
        });
      });

      describe('of <DropdownOption> type but not unique values', () => {
        it('returns error', () => {
          spyOn(PropTypes.node, 'isRequired').and.returnValue(null);

          const validationResult = Dropdown.propTypes.children({
            children: [
              <DropdownOption label={'Valebat clemens acipenser est.'} value={'xx'} />,
              <DropdownOption label={'Valebat clemens acipenser est.'} value={'xx'} />
            ]
          });

          expect(validationResult).toEqual(jasmine.any(Error));
        });
      });
    });

    describe('name', () => {
      describe('PropTypes.string', () => {
        it('returns retrieved error', () => {
          const error = new Error();

          spyOn(PropTypes, 'string').and.returnValue(error);

          const validationResult = Dropdown.propTypes.name();

          expect(validationResult).toBe(error);
        });
      });

      describe('notEmpty', () => {
        it('returns null', () => {
          spyOn(PropTypes, 'string').and.returnValue(null);

          const validationResult = Dropdown.propTypes.name({
            name: 'some name'
          });

          expect(validationResult).toEqual(null);
        });
      });

      describe('isEmpty', () => {
        it('returns null when children is <DropdownOption> ', () => {
          spyOn(PropTypes, 'string').and.returnValue(null);
          const validationResult = Dropdown.propTypes.name({
            children: <DropdownOption value={'adasd'} label={'adsasdad'} />,
            name: null
          });

          expect(validationResult).toEqual(jasmine.any(Error));
        });

        it('returns null when children is <DropdownOptionLink>', () => {
          spyOn(PropTypes, 'string').and.returnValue(null);
          const validationResult = Dropdown.propTypes.name({
            children: <DropdownOptionLink label={'adas'} href={'#'} />,
            name: null
          });

          expect(validationResult).toEqual(null);
        });
      });
    });
  });

  describe('with <DropdownOption>', () => {
    const onChange = () => true;

    it('should be rendered properly', () => {
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'name'}>
          <DropdownOption label={'Label'} value={'some value'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render by default first option as selected', () => {
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'name'}>
          <DropdownOption label={'Label'} value={'asdas'} />
          <DropdownOption label={'Label'} value={'sadasd'} />
          <DropdownOption label={'Label'} value={'xcvxcv'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render third option as selected', () => {
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name="name">
          <DropdownOption label={'Label'} value={'asdas'} />
          <DropdownOption label={'Label'} value={'zxcvc'} />
          <DropdownOption label={'Label'} selected value={'asdasd'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should change selected option when children are updated', () => {
      const getOptions = selectedIndex =>
        ['0', '1', '2'].map(option => (
          <DropdownOption key={option} label={`Option ${option}`} value={option} selected={option === selectedIndex} />
        ));

      const optionsBefore = getOptions();

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'name'}>
          {optionsBefore}
        </Dropdown>
      );

      const optionsAfter = getOptions('2');

      wrapper.setProps({ children: optionsAfter });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.state('currentOption').value).toEqual('2');
    });

    it('should change selected option to first one when children are updated none of them are selected', () => {
      const getOptions = selectedIndex =>
        ['0', '1', '2'].map(option => (
          <DropdownOption key={option} label={`Option ${option}`} value={option} selected={option === selectedIndex} />
        ));

      const optionsBefore = getOptions('2');

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'name'}>
          {optionsBefore}
        </Dropdown>
      );

      const optionsAfter = getOptions();

      wrapper.setProps({ children: optionsAfter });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.state('currentOption').value).toEqual('0');
    });
  });

  describe('with <DropdownOptionLink>', () => {
    const onChange = () => true;

    it('should be rendered properly', () => {
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('when closed', () => {
    const onChange = () => true;

    it('button click should opens it', () => {
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'name'}>
          <DropdownOption label={'Label'} value={'asdad'} />
        </Dropdown>
      );

      expect(wrapper.state().isOpen).toBeFalsy();
      wrapper.find('button').simulate('click');
      expect(wrapper.state().isOpen).toBeTruthy();
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('closeOnOutsideClick() does not change its state', () => {
      const wrapper = mount(
        <Dropdown onChange={onChange} button={<DropdownButton />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      expect(wrapper.state().isOpen).toBeFalsy();

      wrapper.instance().closeOnOutsideClick();

      expect(wrapper.state().isOpen).toBeFalsy();
    });
  });

  describe('when open', () => {
    const onChange = () => true;

    describe('button', () => {
      it('should close it', () => {
        const wrapper = shallow(
          <Dropdown onChange={onChange} button={<button />} name={'name'}>
            <DropdownOption label={'Label'} value={'asdasd'} />
          </Dropdown>
        );

        wrapper.setState({ isOpen: true });
        wrapper.find(DropdownOption).simulate('click');
        expect(wrapper.state().isOpen).toBeFalsy();
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });

    describe('closeOnOutsideClick()', () => {
      it('should close <Dropdown> if clicked on outside of <Dropdown>', () => {
        const wrapper = mount(
          <Dropdown onChange={onChange} button={<DropdownButton />}>
            <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
          </Dropdown>
        );

        wrapper.setState({ isOpen: true });

        expect(wrapper.state().isOpen).toBeTruthy();

        wrapper.instance().root.contains = () => false;

        wrapper.instance().closeOnOutsideClick({});

        expect(wrapper.state().isOpen).toBeFalsy();
      });

      it('should close <Dropdown> if is clicked itself ', () => {
        const wrapper = mount(
          <Dropdown onChange={onChange} button={<DropdownButton />}>
            <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
          </Dropdown>
        );

        wrapper.setState({ isOpen: true });

        expect(wrapper.state().isOpen).toBeTruthy();

        wrapper.instance().closeOnOutsideClick({
          target: wrapper.root
        });

        expect(wrapper.state().isOpen).toBeFalsy();
      });

      it('should not do anything if any child of <Dropdown> is clicked', () => {
        const wrapper = mount(
          <Dropdown onChange={onChange} button={<DropdownButton />}>
            <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
          </Dropdown>
        );

        wrapper.setState({ isOpen: true });

        expect(wrapper.state().isOpen).toBeTruthy();

        wrapper.instance().root.contains = () => true;

        wrapper.instance().closeOnOutsideClick({});

        expect(wrapper.state().isOpen).toBeTruthy();
      });
    });
  });

  describe('should not overwrite sub component callbacks such as', () => {
    const onChange = () => true;

    it('onClick of <DropdownOption>', () => {
      const onClickSpy = jest.fn();

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />} name={'afalsis'}>
          <DropdownOption label={'Eheu, domina!'} onClick={onClickSpy} value={'adasds'} />
        </Dropdown>
      );

      expect(onClickSpy).toHaveBeenCalledTimes(0);

      wrapper.find(DropdownOption).simulate('click');

      expect(onClickSpy).toHaveBeenCalledTimes(1);
    });

    it('onClick of <DropdownOptionLink>', () => {
      const onClickSpy = jest.fn();

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} onClick={onClickSpy} />
        </Dropdown>
      );

      expect(onClickSpy).toHaveBeenCalledTimes(0);

      wrapper.find(DropdownOptionLink).simulate('click');

      expect(onClickSpy).toHaveBeenCalledTimes(1);
    });

    it('onClick of any button', () => {
      const onClickSpy = jest.fn();

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button onClick={onClickSpy} />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      expect(onClickSpy).toHaveBeenCalledTimes(0);

      wrapper.find('button').simulate('click');

      expect(onClickSpy).toHaveBeenCalledTimes(1);
    });

    it('onKeyDown of any button', () => {
      const onKeyDown = jest.fn();

      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button onKeyDown={onKeyDown} />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      expect(onKeyDown).toHaveBeenCalledTimes(0);

      wrapper.find('button').simulate('keyDown');

      expect(onKeyDown).toHaveBeenCalledTimes(1);
    });
  });

  describe('onChange should be', () => {
    describe('called on', () => {
      it('<DropdownOption> click', () => {
        const onChange = jest.fn();

        const wrapper = shallow(
          <Dropdown button={<button />} onChange={onChange} name={'name'}>
            <DropdownOption label={'Eheu, domina!'} value={'asda'} />
            <DropdownOption label={'Eheu, domina!'} value={'test'} />
          </Dropdown>
        );

        wrapper
          .find(DropdownOption)
          .at(1)
          .simulate('click');

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith({
          name: '',
          value: 'test',
          label: 'Eheu, domina!',
          disabled: false,
          selected: false,
          onClick: jasmine.any(Function),
          showPopupOnHover: false,
          className: '',
          icon: ''
        });

        wrapper
          .find(DropdownOption)
          .at(1)
          .simulate('click');
      });
    });

    describe('not be called on', () => {
      it('<DropdownOptionLink> click', () => {
        const onChange = jest.fn();

        const wrapper = shallow(
          <Dropdown button={<button />} onChange={onChange}>
            <DropdownOptionLink label={'Eheu, domina!'} href={'first'} />
            <DropdownOptionLink label={'Eheu, domina!'} href={'second'} />
          </Dropdown>
        );

        wrapper
          .find(DropdownOptionLink)
          .at(1)
          .simulate('click');
        wrapper
          .find(DropdownOptionLink)
          .at(0)
          .simulate('click');

        expect(onChange).toHaveBeenCalledTimes(0);
      });

      it('on button keyDown when the <Dropdown> is open and key is neither arrow up nor arrow down', () => {
        const onChange = jest.fn();

        const wrapper = shallow(
          <Dropdown button={<button />} onChange={onChange} name={'some name'}>
            <DropdownOption label={'Eheu, domina!'} value={'first'} />
            <DropdownOption label={'Eheu, domina!'} value={'second'} />
          </Dropdown>
        );

        wrapper.setState({ isOpen: true });

        // other key than arrow up or arrow down
        wrapper.find('button').simulate('keyDown', { preventDefault: () => {}, keyCode: 55 });

        expect(onChange).not.toHaveBeenCalled();
      });

      it('on button keyDown when the <Dropdown> is closed and key is arrow down', () => {
        const onChange = jest.fn();

        const wrapper = shallow(
          <Dropdown button={<button />} onChange={onChange} name={'name'}>
            <DropdownOption label={'Eheu, domina!'} value={'first'} />
            <DropdownOption label={'Eheu, domina!'} value={'second'} />
          </Dropdown>
        );

        wrapper.find('button').simulate('keyDown', { preventDefault: () => {}, keyCode: 40 });

        expect(onChange).not.toHaveBeenCalled();
      });
    });
  });

  describe('should add proper class name', () => {
    const onChange = () => true;

    it('when className prop was passed', () => {
      const wrapper = mount(
        <Dropdown onChange={onChange} button={<DropdownButton className="foo" />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      const expectedResultNodes = wrapper.find('.foo');
      const unexpectedResultNodes = wrapper.find('.sc-btn-default');

      expect(expectedResultNodes.length).toBeTruthy();
      expect(unexpectedResultNodes.length).toBeFalsy();
    });

    it('when className prop was not passed', () => {
      const wrapper = mount(
        <Dropdown onChange={onChange} button={<DropdownButton />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      const resultNodes = wrapper.find('.sc-btn-default');

      expect(resultNodes.length).toBeTruthy();
    });
  });

  describe('should add and remove event listener onClick "closeOnOutsideClick" callback from document', () => {
    it('when mounted and unmounted', () => {
      const onChange = () => true;

      spyOn(window.document, 'addEventListener');
      spyOn(window.document, 'removeEventListener');

      const wrapper = mount(
        <Dropdown onChange={onChange} button={<DropdownButton />}>
          <DropdownOptionLink label={'Eheu, domina!'} href={'afalsis'} />
        </Dropdown>
      );

      const eventCallback = wrapper.instance().closeOnOutsideClick;

      expect(window.document.addEventListener).toHaveBeenCalledWith('click', eventCallback);
      expect(window.document.removeEventListener).not.toHaveBeenCalled();

      wrapper.unmount();

      expect(window.document.removeEventListener).toHaveBeenCalledWith('click', eventCallback);
    });
  });

  describe('with text inside button prop', () => {
    const buttonText = "Hi, I'm button's text";

    it('should render text if no <DropdownOption> is pre-selected', () => {
      const wrapper = shallow(
        <Dropdown button={<button>{buttonText}</button>}>
          <DropdownOption label={'Eheu, domina!'} value={'afalsis'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render pre-selected option label nevertheless', () => {
      const wrapper = shallow(
        <Dropdown button={<button>{buttonText}</button>}>
          <DropdownOption selected label={'Eheu, domina!'} value={'afalsis'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('this text will be changed if some DD option will be selected', () => {
      const onChange = jest.fn();
      const wrapper = shallow(
        <Dropdown onChange={onChange} button={<button>{buttonText}</button>}>
          <DropdownOption label={'Eheu, domina!'} value={'afalsis'} />
          <DropdownOption label={'domina, Eheu!'} value={'afalsis'} />
        </Dropdown>
      );

      expect(wrapper.find('button').text()).toBe(buttonText);
      expect(wrapper.state().currentOption.label).toBe('Eheu, domina!');

      wrapper.update();
      wrapper.instance().forceUpdate();
      wrapper
        .find(DropdownOption)
        .at(1)
        .simulate('click');

      expect(onChange).toHaveBeenCalledTimes(1);
      expect(wrapper.state().currentOption.label).toBe('domina, Eheu!');
      expect(wrapper.find('button').text()).toBe('domina, Eheu!');
    });
  });

  describe('with withSuffix prop', () => {
    it('should add suffix and proper suffix wrapper classname', () => {
      const suffixData = {
        suffix: 'suf',
        suffixWrapperClassName: 'some-class'
      };
      const buttonText = "Hi, I'm button's text";

      const wrapper = shallow(
        <Dropdown withSuffix={suffixData} button={<button>{buttonText}</button>}>
          <DropdownOption label={'Eheu, domina!'} value={'afalsis'} />
        </Dropdown>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<DropdownOption>', () => {
  describe('prop isChecked', () => {
    it('when false it does not render sc-focus class', () => {
      const wrapper = shallow(<DropdownOption label={'some label'} name={'some name'} value={'some value'} />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('when true it renders sc-focus class', () => {
      const onChange = () => true;

      const wrapper = shallow(
        <DropdownOption onChange={onChange} selected label={'some label'} name={'some name'} value={'some value'} />
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('should add and remove title attribute to label', () => {
    it('when showPopupOnHover prop is true', () => {
      const label = 'show this on hover';

      const wrapper = mount(<DropdownOption label={label} href={'https://'} showPopupOnHover />);

      const labelElement = wrapper.find('label');

      expect(labelElement.length).toBeTruthy();
      expect(labelElement.getDOMNode().getAttribute('title')).toEqual(label);
    });

    it('when showPopupOnHover prop is false', () => {
      const wrapper = mount(<DropdownOption label={'some label text'} href={'https://'} />);

      const labelElement = wrapper.find('label');

      expect(labelElement.length).toBeTruthy();
      expect(labelElement.getDOMNode().getAttribute('title')).toBeNull();
    });
  });
});

describe('<DropdownOptionLink>', () => {
  it('renders', () => {
    const wrapper = shallow(<DropdownOptionLink label={'some label'} href={'#'} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
