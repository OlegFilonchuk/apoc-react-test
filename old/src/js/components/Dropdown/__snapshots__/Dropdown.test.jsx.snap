// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown> when closed button click should opens it 1`] = `
<div
  className="sc-dropdown sc-open"
>
  <button
    isExpanded={true}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Label
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="asdad"
    />
  </ul>
</div>
`;

exports[`<Dropdown> when open button should close it 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Label
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="asdasd"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOption> should be rendered properly 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Label
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="some value"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOption> should change selected option to first one when children are updated none of them are selected 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Option 0
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$0"
      label="Option 0"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="0"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$1"
      label="Option 1"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="1"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$2"
      label="Option 2"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="2"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOption> should change selected option when children are updated 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Option 2
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$0"
      label="Option 0"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="0"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$1"
      label="Option 1"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="1"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".$2"
      label="Option 2"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="2"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOption> should render by default first option as selected 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Label
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="asdas"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".1"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="sadasd"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".2"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="xcvxcv"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOption> should render third option as selected 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Label
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="asdas"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".1"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={false}
      showPopupOnHover={false}
      value="zxcvc"
    />
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".2"
      label="Label"
      name="name"
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="asdasd"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with <DropdownOptionLink> should be rendered properly 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  />
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOptionLink
      className=""
      disabled={false}
      href="afalsis"
      icon=""
      key=".0"
      label="Eheu, domina!"
      onClick={[Function]}
    />
  </ul>
</div>
`;

exports[`<Dropdown> with text inside button prop should render pre-selected option label nevertheless 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Eheu, domina!
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Eheu, domina!"
      name=""
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="afalsis"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with text inside button prop should render text if no <DropdownOption> is pre-selected 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Hi, I'm button's text
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Eheu, domina!"
      name=""
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="afalsis"
    />
  </ul>
</div>
`;

exports[`<Dropdown> with withSuffix prop should add suffix and proper suffix wrapper classname 1`] = `
<div
  className="sc-dropdown"
>
  <button
    isExpanded={false}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Hi, I'm button's text
  </button>
  <ul
    className="sc-dropdown-menus"
  >
    <DropdownOption
      className=""
      disabled={false}
      icon=""
      key=".0"
      label="Eheu, domina!"
      name=""
      onClick={[Function]}
      selected={true}
      showPopupOnHover={false}
      value="afalsis"
    />
  </ul>
  <div
    className="some-class"
  >
    suf
  </div>
</div>
`;

exports[`<DropdownOption> prop isChecked when false it does not render sc-focus class 1`] = `
<li
  className="dropdown-option "
>
  <label
    className=""
  >
    <span
      className=" icon"
    />
    some label
    <input
      data-test-element="option-some value"
      defaultChecked={false}
      disabled={false}
      name="some name"
      onClick={[Function]}
      readOnly="readOnly"
      role="menu"
      type="radio"
      value="some value"
    />
  </label>
</li>
`;

exports[`<DropdownOption> prop isChecked when true it renders sc-focus class 1`] = `
<li
  className="dropdown-option "
>
  <label
    className="sc-focus"
  >
    <span
      className=" icon"
    />
    some label
    <input
      data-test-element="option-some value"
      defaultChecked={true}
      disabled={false}
      name="some name"
      onClick={[Function]}
      readOnly="readOnly"
      role="menu"
      type="radio"
      value="some value"
    />
  </label>
</li>
`;

exports[`<DropdownOptionLink> renders 1`] = `
<li
  className="dropdown-option-link"
>
  <a
    className=""
    disabled={true}
    href="#"
    onClick={[Function]}
  >
    <span
      className=" icon"
    />
    some label
  </a>
</li>
`;
